docs/src/upgrading.md

```diff
@@ -0,0 +1,104 @@
+# Upgrading
+
+JLSO.jl will automatically upgrade older versions of the file format when you call `JLSO.load` or `read`.
```

Comment:
Seems dangerous. This is modifying the file on disk automatically?

Reply:
No, `JLSO.load` and `read` will only upgrade the in memory. The purpose of `JLSO.upgrade` is to explicit upgrade the files on disk to make future loading or loading of batches of files faster. The only way to overwrite existing files it to call `JLSO.upgrade` and pass the same path for both `src` and `dest`.

Reply:
It should probably not say "upgrade" for when it loads a old version.
not on the docs at least.

Reply:
That's what it says in the logs though.

Reply:
Sounds good to me!